pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -204,9 +205,24 @@ func makeVirtualServiceRoute(hosts sets.String, http *v1alpha1.HTTPIngressPath,
 		}
 	}
 
+	var rewrite *istiov1alpha3.HTTPRewrite
+	if http.RewriteHost != "" {
+		rewrite = &istiov1alpha3.HTTPRewrite{
+			Authority: http.RewriteHost,
+		}
+
+		weights = []*istiov1alpha3.HTTPRouteDestination{{
+			Weight: 100,
+			Destination: &istiov1alpha3.Destination{
+				Host: gatewayServiceURLFromContext(ctx, visibility),
```

Comment:
Could we always route the traffic to cluster local gateway?

It is likely that users will configure external Gateway to do HTTPS, authentication, or authorization, etc. So It is possible that the external Gateway won't accept the redirected request that has different host.
(not a blocker for this PR, I am fine to do it in a separate PR).

Reply:
yeah, sounds like that may be better. Agree let's get this landed and do it as a follow-on, though, if you're ok with that